lib/services/access_checker.go

```diff
@@ -470,6 +463,29 @@ func (a *accessChecker) checkAllowedResources(r AccessCheckable) error {
 	return trace.AccessDenied("access to %v denied, not in allowed resource IDs", r.GetKind())
 }
 
+// mapsToURCResource matches requested resource with its respective
+// resource type stored in the unified resource cache.
+func mapsToURCResource(requestedR types.ResourceID, r AccessCheckable) bool {
+	if requestedR.Name != r.GetName() {
+		return false
+	}
+	// If the allowed resource has `Kind=types.KindKubePod` or any other
+	// Kubernetes supported kinds - types.KubernetesResourcesKinds-, we allow the user to
+	// access the Kubernetes cluster that it belongs to.
+	// At this point, we do not verify that the accessed resource matches the
+	// allowed resources, but that verification happens in the caller function.
+	if slices.Contains(types.KubernetesResourcesKinds, requestedR.Kind) {
+		return r.GetKind() == types.KindKubernetesCluster
+	}
+	// Identity Center account is stored as KindApp kind and
+	// KindIdentityCenterAccount subKind in the unified resource cache.
+	if requestedR.Kind == types.KindIdentityCenterAccount {
```

Comment:
Do we need to check that `r`'s major kind is app as well?

Reply:
Good point, and will be safer - https://github.com/gravitational/teleport/pull/55922/commits/330a8aa55467e9dc3d272772356a1c71ce43bdf6#diff-fde01fb35b1a517643fb3a7ace71c2346d688e90c78e3c3fa5cf51d491ae2649R483